modal/volume.py

```diff
@@ -35,8 +40,17 @@ async def reload(self):
 
         Reloading will fail if there are open files for the volume.
         """
-        req = api_pb2.VolumeReloadRequest(volume_id=self.object_id)
-        _ = await retry_transient_errors(self._client.stub.VolumeReload, req)
+        await self._do_reload()
+
+    async def _do_reload(self, lock=True):
+        if lock:
```

Comment:
Easy* conditional `async with`s with seems to be a Python 3.10+ feature :(

```
async with self._lock if lock else contextlib.nullcontext():
  ...
```

*: You could make your own async null context manager, but that remedy seems worse than the issue it solves.

Reply:
It's actually not too bad, and we have one from before in `client_test.cli_test` (`noop_async_context_manager`) - you could move it out into the modal_utils.async_utils module or something and reuse it in both places:

```
@asynccontextmanager
async def asyncnullcontext(*args, **kwargs):
    yield
```